sunpy/map/mapbase.py

```diff
@@ -611,14 +611,19 @@ def heliographic_longitude(self):
         heliographic_longitude = self.meta.get('hgln_obs', None)
 
         if heliographic_longitude is None:
-            if self._default_heliographic_longitude is None:
-                warnings.warn_explicit(
-                    "Missing metadata for heliographic longitude: "
-                    "assuming longitude of 0 degrees",
-                    Warning, __file__,
-                    inspect.currentframe().f_back.f_lineno)
-                self._default_heliographic_longitude = 0
-            heliographic_longitude = self._default_heliographic_longitude
+            if self.meta.get('crln_obs', None) is not None:
+                heliographic_longitude = self.meta.get('crln_obs', None) - get_sun_L0(self.date)
```

Comment:
If we make it to this line we know the value exists, so we should access it like this:
```suggestion
                heliographic_longitude = self.meta['crln_obs'] - get_sun_L0(self.date)
```